code/script.js

```diff
@@ -32,16 +41,145 @@ const showMessage = (message, sender) => {
+
+}
+
+//INPUT FOR QUESTION 4
+const getNamething = (event) => {
+  inputWrapper.innerHTML = `
+  <div class="input-wrapper2" id="input-wrapper2">
+  <form id="name-form2">
+    <input id="name-input2" type="text" />
+    <button id="sendName" class="send-btn2" type="submit">
+      Send
+    </button>
+  </form>
+</div>
+  `
+
+  document.getElementById('name-form2').addEventListener('submit', handleNameInput)
+
 }
 
+// NAME INPUT HANDLING
+const handleNameInput = (event) => {
+
+  const nameInput = document.getElementById('name-input2')
+
+  event.preventDefault();
+  const userName = nameInput.value
+  showMessage(userName, 'user')
+  nameInput.value = ""
+  setTimeout(() => showMessage
+  (`So sorry ${userName}, that was a bad ice breaker.. I am a bit nervous about this job. I am just an actress after all.`, 'bot'), 2000)
+  setTimeout(() => showMessage
+  (`Anyway, now that we are talking about famous love stories.. Do you know what Jay-Z called Beyoncé after their engagement?`, 'bot'), 5000)
+  setTimeout(() => beyonceResponse(event), 5000)
+}
+
+// QUESTION 5
+const beyonceResponse = (event) => {
+  inputWrapper.innerHTML = buttonSelectMain
+  document
+  .getElementById('feyonceButton')
+  .addEventListener('click', () => {
+    showMessage(`Feyoncé`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => showMessage(`Haha! Funny. Seems like you are better at this than I am. So I am just gonna give you the support material they gave me, and you can take over. Ok?`, 'bot'), 1000)
+    setTimeout(() => showMessage(`Just click the Download button and you'll have it.`, 'bot'), 3000)
+    setTimeout(() => punsT(event), 5000)
+  })
+  document
+  .getElementById('lazyButton')
+  .addEventListener('click', () => {
+    showMessage(`Sigh.. This pun is just Lay-Z`, 'user')
+    inputWrapper.innerHTML = ''
+    setTimeout(() => showMessage(`I guess you are better at this than I am then.. So I am just gonna give you the support material they gave me, and you can take over. Ok?`, 'bot'), 1000)
+    setTimeout(() => showMessage(`Just click the Download button and you'll have it.`, 'bot'), 3000)
+    setTimeout(() => punsT(event), 5000)
+  })
+}
+
+let mainChoice = () => {
+  if (mainChoice === 'feyonceButton') {
+    setTimeout(() => showMessage
+  (`Haha! Funny. Seems like you are better at this than I am. So I am just gonna give you the support material they gave me, and you can take over. Ok?`, 'bot'), 3000)
+  }
+  else if (mainChoice === 'lazyButton') {
+    setTimeout(() => showMessage
+  (`I guess you are better at this than I am then.. So I am just gonna give you the support material they gave me, and you can take over. Ok?`, 'bot'), 3000)
+  }
+}
+
+const punsT = () => {
+  inputWrapper.innerHTML = `
+  <a id="downloadtag" href="https://cdn.iwastesomuchtime.com/December-10-2012-20-12-11-61851310094279099942183677218n.jpg" hidden download></a>
+
+  <button onclick="document.getElementById('downloadtag').click()">Download</button>`
+}
```

Comment:
Haha love this! Maybe this function could be called `showDownloadButton` or something?